src/FlowScene.cpp

```diff
@@ -166,26 +180,46 @@ FlowScene::
 restoreNode(QJsonObject const& nodeJson)
 {
   QString modelName = nodeJson["model"].toObject()["name"].toString();
-
   auto dataModel = registry().create(modelName);
 
   if (!dataModel)
     throw std::logic_error(std::string("No registered model with name ") +
                            modelName.toLocal8Bit().data());
-
   auto node = std::make_unique<Node>(std::move(dataModel));
   auto ngo  = std::make_unique<NodeGraphicsObject>(*this, *node);
   node->setGraphicsObject(std::move(ngo));
-
   node->restore(nodeJson);
 
   auto nodePtr = node.get();
   _nodes[node->id()] = std::move(node);
-
   nodeCreated(*nodePtr);
   return *nodePtr;
 }
 
+QUuid FlowScene::pasteNode(QJsonObject &nodeJson) {
+  QString modelName = nodeJson["model"].toObject()["name"].toString();
+  auto dataModel = registry().create(modelName);
+
+  if (!dataModel)
```

Comment:
Consider this scenario:

Scene S1 with a set of registered models M1.
Scene S2 with a set of registered models M2.

where M1 != M2 or M2 subset of M1.

If you try to copy and paste a node from S1 to S2 which not belong to M2 it will throw.
A more correct approach would be returning with doing anything with a simple warning message.